crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -176,10 +176,11 @@ pub(crate) async fn analyze_ecmascript_module(
     let mut analysis = AnalyzeEcmascriptModuleResultBuilder::new();
     let path = source.path();
 
+    // Is this a typescript file that requires analzying type references?
     let is_typescript = match &*ty {
-        EcmascriptModuleAssetType::Typescript
+        EcmascriptModuleAssetType::TypescriptWithTypes
         | EcmascriptModuleAssetType::TypescriptDeclaration => true,
-        EcmascriptModuleAssetType::Ecmascript => false,
+        EcmascriptModuleAssetType::Typescript | EcmascriptModuleAssetType::Ecmascript => false,
     };
 
     let parsed = parse(source, ty, transforms);
```

Comment:
Nit: there are some references to `is_typescript` in the code below that are not actually used, it's just a parameter that exists to be passed to other code that also doesn't use it and passes down.